maestro-android/src/main/java/dev/mobile/maestro/handlers/AbstractSettingHandler.kt

```diff
@@ -0,0 +1,42 @@
+package dev.mobile.maestro.handlers
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+
+abstract class AbstractSettingHandler(private val context: Context, private val permissions: List<String>) {
+    fun enable(): Boolean {
+        Log.d(TAG, "Enabling $settingDescription")
+        return if (!hasPermissions()) {
+            false
+        } else setState(true)
+    }
+
+    fun disable(): Boolean {
+        Log.d(TAG, "Disabling $settingDescription")
+        return if (!hasPermissions()) {
+            false
+        } else setState(false)
+    }
+
+    protected fun hasPermissions(): Boolean {
```

Comment:
Is this really important ? I think permissions required would be auto granted while installation because we include -g. You can check though

Reply:
in our case yes, from the appium code I see that they added it specifically for location and animation settings.

do you think it is an excessive check?

Reply:
I think this could be implicitly granted. I would suggest experimenting it and then decide to include/not include.

Reply:
I'd keep it still, since it is a minor thing and this implementation was inspired by appium, pretty sure they had it there for a reason